src/main/java/org/wildfly/security/password/impl/PasswordFactorySpiImpl.java

```diff
@@ -274,9 +286,21 @@ protected Password engineGeneratePassword(final String algorithm, final KeySpec
                 }
             }
             case ALGORITHM_BSD_CRYPT_DES: {
-                if (keySpec instanceof BSDUnixDESCryptPasswordSpec) {
+                if (keySpec instanceof IteratedSaltedHashPasswordSpec) {
                     try {
-                        return new BSDUnixDESCryptPasswordImpl((BSDUnixDESCryptPasswordSpec) keySpec);
+                        return new BSDUnixDESCryptPasswordImpl((IteratedSaltedHashPasswordSpec) keySpec);
+                    } catch (IllegalArgumentException | NullPointerException e) {
+                        throw new InvalidKeySpecException(e);
+                    }
+                } if (keySpec instanceof SaltedHashPasswordSpec) {
+                    try {
+                        return new BSDUnixDESCryptPasswordImpl((SaltedHashPasswordSpec) keySpec);
+                    } catch (IllegalArgumentException | NullPointerException e) {
+                        throw new InvalidKeySpecException(e);
+                    }
+                } if (keySpec instanceof SaltedHashPasswordSpec) {
```

Comment:
This will always be false - I think it was double-pasted?  IDEA should have underlined this, if you use IDEA.